### PR TITLE
Feature/in site notifications

### PR DESCRIPTION
**Pull request by Cucurrucho**

### CODE REVIEW

File: app/Http/Requests/User/SendMessageRequest.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+class SendMessageRequest extends FormRequest
+{
+    /**
+     * Determine if the user is authorized to make this request.
+     *
+     * @return bool
+     */
+    public function authorize()
+    {
+        return true;
```
Comment (LCD344) [inline app/Http/Requests/User/SendMessageRequest.php]: authorization here should have a logged in user. or something like that

File: app/Listeners/User/GetMessage.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+     * Handle the event.
+     *
+     * @param  object  $event
+     * @return void
+     */
+    public function handle(MessageSent $event)
+    {
+        $users = $event->conversation->users;
+        $recipient = $users->firstWhere('id','!=', $event->sender->id);
+        $recipient->notify(new Conversation($event->sender,$event->message));
```
Comment (LCD344) [inline app/Listeners/User/GetMessage.php]: The class name shouldnt be conversation - we're not starting a new conversation here - just a message

File: app/Models/User.php [hunk 3]
```diff
@@ -161,4 +162,40 @@ public function delete(): ?bool {
...
+	    return $this->hasMany(Message::class);
+    }
+
+
+    /**
+     * @param User $user
+     * @return mixed
+     */
+    public function getConversationWith(User $user){
+        /*return $this->conversations()->whereHas('users', function ($query) use ($user){
```
Comment (LCD344) [inline app/Models/User.php]: remove comment

File: app/Notifications/Match/JoinMatchRequest.php [hunk 4]
```diff
@@ -77,4 +79,11 @@ public function toArray($notifiable): array {
 			//
 		];
 	}
+
+	public function toConversation($notifiable)
+    {
```
Comment (LCD344) [inline app/Notifications/Match/JoinMatchRequest.php]: adda referance to the match here - so we can show a link to which match he talks about

File: app/Notifications/User/Conversation.php [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+use App\Channels\ConversationChannel;
+use App\Models\Message;
+use App\Models\User;
+use Illuminate\Bus\Queueable;
+use Illuminate\Notifications\Notification;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Notifications\Messages\MailMessage;
+
+class Conversation extends Notification
```
Comment (LCD344) [inline app/Notifications/User/Conversation.php]: this class name is wrong - this is not a new conversation, its a message

File: resources/assets/js/components/Elements/Conversation.vue [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+            },
+        },
+
+        methods: {
+            isCurrent(userId) {
+                return this.currentUser == userId;
+            },
+
+            updateConversation(response, data) {
+                this.messages.push({'text': response.message, 'user_id': this.currentUser});
```
Comment (LCD344) [inline resources/assets/js/components/Elements/Conversation.vue]: we need to add the time here as well

File: resources/assets/js/components/Elements/Conversation.vue [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+        <div class="card-body">
+            <div v-for="message in messages" :class="{'text-right': isCurrent(message.user_id)}"
+                 class="message-wrapper">
+            <span class="message"
+                  :class="[isCurrent(message.user_id) ? 'current-user' : 'sender']">{{message.text}}</span>
+                <span class="time" :class="[isCurrent(message.user_id) ? 'current-user-time' : 'sender-time']">{{message.date}} {{message.time}}</span>
+            </div>
+        </div>
+        <div class="card-footer">
+            <ajax-form event="sent" ref="form"
```
Comment (LCD344) [inline resources/assets/js/components/Elements/Conversation.vue]: we need to clear the input of the form once the button is clicked

File: app/Http/Requests/User/SendMessageRequest.php (hunk 1 above)
Comment (Cucurrucho) [reply app/Http/Requests/User/SendMessageRequest.php]: done

File: app/Models/User.php (hunk 3 above)
Comment (Cucurrucho) [reply app/Models/User.php]: done

File: app/Notifications/User/Conversation.php (hunk 5 above)
Comment (Cucurrucho) [reply app/Notifications/User/Conversation.php]: done

File: app/Listeners/User/GetMessage.php (hunk 2 above)
Comment (Cucurrucho) [reply app/Listeners/User/GetMessage.php]: done

File: resources/assets/js/components/Elements/Conversation.vue (hunk 6 above)
Comment (Cucurrucho) [reply resources/assets/js/components/Elements/Conversation.vue]: done

**Comment by Cucurrucho** [comment]
done

File: resources/assets/js/components/Elements/Conversation.vue (hunk 7 above)
Comment (Cucurrucho) [reply resources/assets/js/components/Elements/Conversation.vue]: done

**Pull request merged**